src/contracts/core/AllocationManager.sol

```diff
@@ -198,13 +201,68 @@ contract AllocationManager is
         uint16[] calldata numToClear
     ) external onlyWhenNotPaused(PAUSED_MODIFY_ALLOCATIONS) {
         require(strategies.length == numToClear.length, InputArrayLengthMismatch());
-        require(delegation.isOperator(operator), OperatorNotRegistered());
 
         for (uint256 i = 0; i < strategies.length; ++i) {
             _clearDeallocationQueue({operator: operator, strategy: strategies[i], numToClear: numToClear[i]});
         }
     }
 
+    /// @inheritdoc IAllocationManager
+    function registerForOperatorSets(
+        RegisterParams calldata params
+    ) external onlyWhenNotPaused(PAUSED_OPERATOR_SET_REGISTRATION_AND_DEREGISTRATION) {
+        // Check that the operator exists
+        require(delegation.isOperator(msg.sender), InvalidOperator());
+
+        for (uint256 i = 0; i < params.operatorSetIds.length; i++) {
+            // Check the operator set exists and the operator is not currently registered to it
+            OperatorSet memory operatorSet = OperatorSet(params.avs, params.operatorSetIds[i]);
+            require(_operatorSets[operatorSet.avs].contains(operatorSet.id), InvalidOperatorSet());
+            require(!_isRegistered(msg.sender, operatorSet), AlreadyMemberOfSet());
+
+            // Add operator to operator set
+            registeredSets[msg.sender].add(operatorSet.key());
+            _operatorSetMembers[operatorSet.key()].add(msg.sender);
+            emit OperatorAddedToOperatorSet(msg.sender, operatorSet);
+
+            // Mark the operator registered
+            registrationStatus[msg.sender][operatorSet.key()].registered = true;
+        }
+
+        // Call the AVS to complete registration. If the AVS reverts, registration will fail.
+        getAVSRegistrar(params.avs).registerOperator(msg.sender, params.operatorSetIds, params.data);
+    }
+
+    /// @inheritdoc IAllocationManager
+    function deregisterFromOperatorSets(
+        DeregisterParams calldata params
+    ) external onlyWhenNotPaused(PAUSED_OPERATOR_SET_REGISTRATION_AND_DEREGISTRATION) {
+        require(msg.sender == params.operator || msg.sender == params.avs, InvalidCaller());
+
+        for (uint256 i = 0; i < params.operatorSetIds.length; i++) {
+            // Check the operator set exists and the operator is registered to it
+            OperatorSet memory operatorSet = OperatorSet(params.avs, params.operatorSetIds[i]);
+            require(_operatorSets[params.avs].contains(operatorSet.id), InvalidOperatorSet());
```

Comment:
Noting since operatorSetIds cant be deleted, this opSetId existing check is kind of implied if the operator was able to register in the first place. Fine to leave as is but I don't think it's required

Reply:
I removed a bunch of extraneous checks but i like this one for clarity. Good point though!